packages/rlc-common/src/interfaces.ts

```diff
@@ -12,6 +12,11 @@ export interface RLCModel {
   importSet?: Map<ImportKind, Set<string>>;
   helperDetails?: HelperFunctionDetails;
   urlInfo?: UrlInfo;
+  telemetryOptions?: TelemetryOptions;
+}
+
+export interface TelemetryOptions {
+  customRequestIdHeaderName?: string;
 }
```

Comment:
why not import this TelemetryOptions from core-rest-pipeline ?

Reply:
I don't think adding the dependency `core-rest-pipeline` in rlc-common is a good idea, they are different things. One is the interface for client option, the other is the option for codegen to build factory method. Maybe I could try to think out a more specific name here.